cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -357,6 +358,30 @@ def __init__(self, sys_cfg, distro, paths):
         DataSourceNoCloud.__init__(self, sys_cfg, distro, paths)
         self.supported_seed_starts = ("http://", "https://")
 
+    def ds_detect(self):
+        """NoCloud requires "nocloud-net" as the way to specify
+        seeding from an http(s) address. This diverges from all other
+        datasources in that it does a kernel commandline match on something
+        other than the datasource dsname for only DEP_NETWORK.
+        """
+        cmdline = util.get_cmdline()
+        ds_parse_0 = re.search(r"ds=([a-zA-Z]+)(\s|$)", cmdline)
+        ds_parse_1 = re.search(r"ci\.ds=([a-zA-Z]+)(\s|$)", cmdline)
+        ds_parse_2 = re.search(r"ci\.datasource=([a-zA-Z]+)(\s|$)", cmdline)
+        ds = ds_parse_0 or ds_parse_1 or ds_parse_2
+        deprecated = ds_parse_1 or ds_parse_2
+        if deprecated:
+            util.deprecate(
+                deprecated=(
+                    f"Defining the datasource on the commandline using "
+                    f"ci.ds={ds} or ci.datasource={ds}"
```

Comment:
This is actually printing the re.search results, not the datasource name.
Let's just tell them what they sent. no guesswork needed.
```suggestion
                    f"Defining the datasource on the commandline using "
                    f"{ds.group().strip()}"
```

Reply:
thanks! done